llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1573,6 +1573,21 @@ void SPIRVToLLVM::updateDebugLoc(SPIRVValue *bv, Function *f) {
   }
 }
 
+// =====================================================================================================================
+// This function will be removed when LLPC switch to opaque pointers
+//
+// Find function in the module with specific type and starting with name.
+Function *SPIRVToLLVM::findFunctionInModule(string name, FunctionType *funcType) {
+
+  for (Function &func : m_m->functions()) {
+    if (func.getName().startswith(name) && func.getFunctionType() == funcType) {
+      return &func;
+    }
+  }
+
+  return nullptr;
+}
```

Comment:
A bit unorthodox, but I suppose we can live with it briefly.

**However**, please add a fast path that uses the `Module`'s own lookup functions once and checks the type. Those lookups use a hash map and are therefore much faster than iteration. If you don't get any result from the lookup, you can immediately return null. If you get an incorrectly typed result, you can go into the loop.

Reply:
Thanks ! I will move this to draft for now.